### PR TITLE
Feature/create food card component

### PR DESCRIPTION
**Pull request by CBinh710**
### Create food card component
https://github.com/users/CBinh710/projects/2/views/1?pane=issue&itemId=72043591
image

### CODE REVIEW

File: practice-one/src/components/food-card.html [hunk 1]
```diff
@@ -0,0 +1,14 @@
+<div class="card-container">
+  <div class="card-filter">
+      <div class="food-image"><img src="./assets/images/kebab.png" alt=""></div>
+      <div class="food-price"><p>10$</p></div>
+  </div>
+  <div class="food-rating"><img src="./assets/images/kebab-rating.png" alt=""></div>
+  <p class="food-card-header">Kebab</p>
+  <p class="food-card-text">Lorem, ipsum dolor sit amet consectetur adipisicing elit. Sit repellendus repudiandae quod error? Distinctio provident enim odio sunt, ea, ullam eligendi dolore non saep.</p>
```
Comment (hungdo-agilityio) [inline practice-one/src/components/food-card.html]: This is not correct, we can pass price, name of the food, description of the disk like we pass `text` of `button.html`

Comment (CBinh710) [reply practice-one/src/components/food-card.html]: i fixed it

File: practice-one/src/assets/images/kebab-rating.png
Comment (hungdo-agilityio) [inline practice-one/src/assets/images/kebab-rating.png:1]: average rating like `(4.5)` in this image, should not including in image, this one should be pass like other prop

File: practice-one/src/components/food-card.html [hunk 2]
```diff
@@ -0,0 +1,14 @@
+<div class="card-container">
+  <div class="card-filter">
+      <div class="food-image"><img src="{{image}}" alt=""></div>
```
Comment (hungdo-agilityio) [inline practice-one/src/components/food-card.html]: Missing alt attribute, please recheck why alt is important

File: practice-one/src/index.html [hunk 3]
```diff
@@ -9,6 +9,15 @@
...
 <body class="container">
   <include src="./components/header.html"></include>
+  <include src="./components/food-card.html" 
+  locals='{
+    "image": "./assets/images/kebab.png",
+    "rating": "./assets/images/kebab-rating.png",
+    "price":"10$",
+    "header":"Kebab",
+    "text": "Lorem, ipsum dolor sit amet consectetur adipisicing elit. Sit repellendus repudiandae quod error? Distinctio provident enim odio sunt, ea, ullam eligendi dolore non saep."
+  }'
```
Comment (hungdo-agilityio) [inline practice-one/src/index.html]: Please check indent here

File: practice-one/src/components/food-card.html (hunk 2 above)
Comment (hungdo-agilityio) [reply practice-one/src/components/food-card.html]: Rename class this should be `food-image-wrapper` or `food-image-container` to more transparent, please recheck for all

File: practice-one/src/styles/components/food-card.css [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+  border: 10px solid var(--color-light);
+  background-color: var(--color-accent);
+  border-radius: 50%;
+  top: 20px;
+  left: 20px;
+}
+
+.food-card-header {
+  color: var(--color-primary);
+  font-family: var(--font-family-bold);
```
Comment (hungdo-agilityio) [inline practice-one/src/styles/components/food-card.css]: use font-weight instead

File: practice-one/src/styles/components/food-card.css [hunk 5]
```diff
@@ -0,0 +1,62 @@
+.card-container {
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  width: 310px;
+  height: 516px;
```
Comment (hungdo-agilityio) [inline practice-one/src/styles/components/food-card.css]: Height of the box with background gradient is not correct, please recheck the design

File: practice-one/src/styles/components/food-card.css [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+  text-align: center;
+  position: absolute;
+  margin: 20px;
+  top: 350px;
+}
+
+.card-button {
+  position: absolute;
+  top: 500px;
+}
```
Comment (hungdo-agilityio) [inline practice-one/src/styles/components/food-card.css]: Please recheck the position of this button, look at your image attach in description, seem like it was wrong

File: practice-one/src/styles/components/food-card.css (hunk 5 above)
Comment (CBinh710) [reply practice-one/src/styles/components/food-card.css]: i fixed it, thanks

File: practice-one/src/components/food-card.html (hunk 2 above)
Comment (CBinh710) [reply practice-one/src/components/food-card.html]: i understood , thanks

File: practice-one/src/styles/components/food-card.css (hunk 4 above)
Comment (CBinh710) [reply practice-one/src/styles/components/food-card.css]: i fixed it

File: practice-one/src/styles/components/food-card.css (hunk 6 above)
Comment (CBinh710) [reply practice-one/src/styles/components/food-card.css]: i checked and fixed it

File: practice-one/src/index.html (hunk 3 above)
Comment (CBinh710) [reply practice-one/src/index.html]: i fixed it

File: practice-one/src/assets/images/kebab-rating.png
Comment (CBinh710) [reply practice-one/src/assets/images/kebab-rating.png:1]: i changed it , thanks

File: practice-one/src/components/food-card.html [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+  </div>
+  <div class="food-price">
+    <p>{{price}}</p>
+  </div>
+  <div class="card-detail">
+    <div class="food-rating">
+      <div class="rating-1"><img src="{{rating1}}" alt="Rating 1"/></div>
+      <div class="rating-2"><img src="{{rating2}}" alt="Rating 2"/></div>
+      <div class="rating-3"><img src="{{rating3}}" alt="Rating 3"/></div>
+      <div class="rating"><img src="{{ratingStar}}" alt="Rating Star"/></div>
```
Comment (hungdo-agilityio) [inline practice-one/src/components/food-card.html]: any case `ratingStar` is different and we must pass another image here

File: practice-one/src/index.html [hunk 8]
```diff
@@ -9,6 +9,19 @@
...
   <include src="./components/header.html"></include>
+  <include src="./components/food-card.html"
+  locals='{
+    "image": "./assets/images/kebab.png",
+    "imageAlt":"Kebab image",
+    "rating1": "./assets/images/rating-1.png",
+    "rating2": "./assets/images/rating-2.png",
+    "rating3": "./assets/images/rating-3.png",
+    "ratingStar": "./assets/images/rating-star.png",
+    "rating": "(4.5)",
```
Comment (hungdo-agilityio) [inline practice-one/src/index.html]: do we need to pass bracket every time, can we just pass `4.5` or `.5` only

File: practice-one/src/styles/components/food-card.css [hunk 9]
```diff
@@ -0,0 +1,98 @@
...
+  font-weight: bolder;
+}
+
+.food-detail {
+  margin: 10px 20px;
+}
+
+.card-button {
+  position: relative;
+  top: 30px;
```
Comment (hungdo-agilityio) [inline practice-one/src/styles/components/food-card.css]: `top: 30px` not make sure button is position half on the card. Use can use percentage here
```
bottom: 0;
transform: translateY(50%)
```

File: practice-one/src/styles/components/food-card.css [hunk 10]
```diff
@@ -0,0 +1,98 @@
...
+}
+
+.rating-2 {
+  z-index: 2;
+  left: 70px;
+}
+
+.rating-3 {
+  z-index: 3;
+  left: 90px;
```
Comment (hungdo-agilityio) [inline practice-one/src/styles/components/food-card.css]: can we use `:nth` pseudo

File: practice-one/src/components/food-card.html (hunk 7 above)
Comment (hungdo-agilityio) [inline practice-one/src/components/food-card.html]: Recheck the `alt`

File: practice-one/src/components/food-card.html [hunk 11]
```diff
@@ -0,0 +1,22 @@
+<div class="food-card-container">
+  <div class="food-card-wrapper">
+    <img src="{{image}}" alt="{{imageAlt}}" />
```
Comment (hungdo-agilityio) [inline practice-one/src/components/food-card.html]: No need to pass `imageAlt` use the `name` instead

File: practice-one/src/index.html (hunk 8 above)
Comment (CBinh710) [reply practice-one/src/index.html]: thanks i fixed it

File: practice-one/src/components/food-card.html (hunk 11 above)
Comment (CBinh710) [reply practice-one/src/components/food-card.html]: Thanks

File: practice-one/src/components/food-card.html (hunk 7 above)
Comment (CBinh710) [reply practice-one/src/components/food-card.html]: i fixed it

File: practice-one/src/styles/components/food-card.css (hunk 10 above)
Comment (CBinh710) [reply practice-one/src/styles/components/food-card.css]: i will try it

File: practice-one/src/styles/components/food-card.css (hunk 9 above)
Comment (CBinh710) [reply practice-one/src/styles/components/food-card.css]: i changed it , best regard

File: practice-one/src/components/food-card.html (hunk 7 above)
Comment (CBinh710) [reply practice-one/src/components/food-card.html]: i understood, thanks

File: practice-one/src/index.html [hunk 12]
```diff
@@ -9,6 +9,19 @@
...
+  locals='{
+    "image": "./assets/images/kebab.png",
+    "imageAlt":"Kebab image",
+    "rating1": "./assets/images/rating-1.png",
+    "rating2": "./assets/images/rating-2.png",
+    "rating3": "./assets/images/rating-3.png",
+    "ratingStar": "./assets/images/rating-star.png",
+    "rating": "4.5",
+    "price":"10$",
+    "name":"Kebab",
```
Comment (hungdo-agilityio) [inline practice-one/src/index.html]: Recheck redundant

**Review by hungdo-agilityio — APPROVED** [review]

**Pull request merged**